src/unittest/test_utilities.cpp

```diff
@@ -253,6 +255,13 @@ void TestUtilities::testUTF8()
 		== "the shovel dug a crumbly node!");
 }
 
+void TestUtilities::testRemoveEscapes()
+{
+	UASSERT(remove_escapes(L"abc\x1bXdef") == L"abcdef");
+	UASSERT(remove_escapes(L"abc\x1b(escaped)def") == L"abcdef");
+	UASSERT(remove_escapes(L"abc\x1b((escaped with parenthesis\\))def") == L"abcdef");
```

Comment:
This tests functionality well, but could you also add some edge cases so I have confidence it'll work?
For example, an incomplete escape sequence at the end of the string, an escape sequence with an unclosed set of parentheses, an escape sequence without a set of parentheses at all, etc.  The point of a test is to try your hardest to break the code, not feed it values you already know work well.